### ISSUE TITLE
Different results of t test depending on dependent variable's measurement level

### ISSUE THREAD

**Issue by asoskic** [jasp-stats/jasp-desktop#2245]
Hi! I am getting different results of independent samples t test (as well as Levene's test), depending on whether the variable has been marked as nominal or ordinal, on one hand, or scale on the other hand.

I have noticed this first with a file imported from SPSS, so I made another file to test this problem further. I have attached it here:
[JASP t test and measurement level.zip](https://github.com/jasp-stats/jasp-desktop/files/1575620/JASP.t.test.and.measurement.level.zip)

**Test dataset description:** The file was created in SPSS, and then imported into JASP. It has an independent variable (Group), a participant ID variable and 6 dependent variables. These dependent variables were created from two original variables, one with and one without missing data. They have been copied into three identical pairs, one pair labelled as having nominal, one ordinal, and one scale level of measurement.
*It probably doesn't matter, but it might for some reason: Group variable doesn't include all cases, only some of them.
*I initially thought the problem might be connected with missing data, so I included that variation in my test dataset - it turned out to be irrelevant.

**Problem description:** When applying independent samples t test (variable Group --> dependent variables), I got matching results for nominal and ordinal variables, but they were different for scale variables. By the way, SPSS gives the same result as I got with scale variables in JASP (regardless of the level of measurement).

* JASP version: 0.8.4.
* OS name and version: Windows 10 Education
* Analysis: T test for independent samples
* Steps to reproduce: Independent samples t test on dependent variables with identical contents, but different level of measurement. I have attached a file with the database I used and results that I got.

**Comment by EJWagenmakers** [comment]
I think this has something to do with the SPSS import. Some observations:
1. When I upload a csv and change the measurement level in JASP, the results stay the same.
2. If you take your variables and you select "descriptives", you will see that they differ. So even though the numbers *look* the same, the really aren't.
3. This becomes clear when you change the measurement level of the columns in JASP -- this changes the data! @FransMeerhoff @boutinb -- this is unwanted behavior, SPSS should not be able to fool JASP this way. 
 
Cheers,
E.J.

**Comment by asoskic** [comment]
Thank you for a quick reply!

This is really strange to me, because the raw numbers seem to be identical, but the descriptives and tests differ indeed. I generated a .csv file for editing, and the numbers in it also looked the same. When I just saved it without making any changes, all of my "nominal" and "ordinal" variables became automatically "scale" variables in JASP and all of the descriptives and test results became identical, while the raw numbers still look the same as before.

**Comment by boutinb** [comment]
@asoskic Can you give me your SPSS file as well?

**Comment by asoskic** [comment]
@boutinb For sure, here you are!
[SPSS t test and measurement level.zip](https://github.com/jasp-stats/jasp-desktop/files/1576126/SPSS.t.test.and.measurement.level.zip)

**Comment by boutinb** [comment]
@asoskic I don't understand how you manage to put nominal & ordinal columns in the Dependent Variables of the t-test, because it only accepts scale columns.
Also if I load your SPSS file in JASP, I can change the nominal & ordinal columns in scale type, without changing the values... Maybe you used first an older version of JASP (I know that I repaired some mistakes some months ago), and you use now the same JASP file. Could you try to load SPSS file with the new version if JASP?

**Comment by asoskic** [comment]
Hi! I think you are right.
I used an older version of JASP (0.8.0.1) at first, but when I noticed this was happening, I installed the latest version to see if the issue would still be there. I tried just now to load the SPSS file into JASP, and the result is that it automatically turns my ordinal variables into nominal and Group variable into ordinal, and I can insert only scale variables into t test.
However, with the .jasp file that I sent you at first, it is perfectly possible to insert nominal and ordinal variables into t test in the JASP version I am now using (8.4.0), so I believe it must be created by the previous version of JASP.
